.circleci/config.yml

```diff
@@ -12,6 +12,7 @@ jobs:
       - run:
           name: Build and test examples
           command: |
+            git submodule update --init
```

Comment:
In the [circleCI documentation](https://circleci.com/docs/configuration-reference/#checkout), they advise to do:
```yml
    steps:
      - checkout
      - run: git submodule sync
      - run: git submodule update --init
      - run:
          name: Build and test examples
          command: |
            mkdir build
            ...
```
I think this is to avoid `git submodules` traces to be inside the build traces. We can keep your proposal, I see no real added values.

Reply:
Having so many steps gives a rather ugly page on CircleCI. To me, this should be part of `checkout`, and indeed [Microsoft](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines) has an option `submodules: true` for checkout, but I don't see anything similar in CircleCI. I guess I could at least merge the 2 submodule commands into 1 run.